Mathlib/Init/Data/Nat/Lemmas.lean

```diff
@@ -190,6 +190,7 @@ protected def strong_rec_on {p : ℕ → Sort u}
   (n : ℕ) (H : ∀ n, (∀ m, m < n → p m) → p n) : p n :=
 Nat.lt_wfRel.wf.fix' H n
 
+@[elabAsElim]
```

Comment:
Why do we need `@[elabAsElim]` here, while the Lean3 version does not have `@[elab_as_eliminator]`?
https://github.com/leanprover-community/lean/blob/741670c439f1ca266bc7fe61ef7212cc9afd9dd8/library/init/data/nat/lemmas.lean#L784-L786

Reply:
From what I can tell Lean 4 seems to be a bit stricter with `elabAsElim`.  Without the attribute I had to give the motive manually.